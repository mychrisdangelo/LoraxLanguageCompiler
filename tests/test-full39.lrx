/* 
 * Author: Chris D'Angelo
 * Test for parent function
 */

int main()
{
	tree <int>t(2);
	tree <int>t2(2);
	t = 1[2, 3[4, 5]];
	t2 = 0[1, 2];
	print("should print 3[4, 5] = ", parent(t%1%0), "\n");
	print("should print null = ", parent(1[2, 3]), "\n");
	print("should print 1[2, 3[4, 5] = ", parent(parent(t%1%0)), "\n");
	t2%0 = parent(parent(t%1%0));
	print("should print 0[1[2, 3[4, 5], null] = ", t2, "\n");
	// will cause assertion failure
	// print("what is this printing? ", parent(parent(t)), "\n");
}
