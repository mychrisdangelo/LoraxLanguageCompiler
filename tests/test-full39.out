should print 3[4, 5] = 3[4[null,null],5[null,null]]
should print null = null
should print 1[2, 3[4, 5] = 1[2[null,null],3[4[null,null],5[null,null]]]
should print 0[1[2, 3[4, 5], null] = 0[1[2[null,null],3[4[null,null],5[null,null]]],2[null,null]]
